src/lib/patch.js

```diff
@@ -23,33 +23,36 @@ import cockpit from 'cockpit';
 
 export class Patch {
     static async patches() {
-        const xml = await cockpit.spawn(["zypper", "--xmlout", "list-patches"], { superuser : "require" });
-
-        const parser = new DOMParser();
-        const xmlDoc = parser.parseFromString(xml, "text/xml");
-
-        const updates = xmlDoc.getElementsByTagName("update");
-
-        return Array.from(updates).map((update) => {
-            return new Patch(
-                update.getAttribute("name"),
-                update.getAttribute("edition"),
-                update.getAttribute("category"),
-                update.getAttribute("severity"),
-                update.getElementsByTagName("summary")[0].textContent,
-                update.getElementsByTagName("description")[0].textContent
-            );
+        // variable for collecting the found patches
+        const patches = [];
+
+        const dbus_client = cockpit.dbus("org.opensuse.transactional_update");
+        const dbus_proxy = dbus_client.proxy("org.opensuse.transactional_update",
```

Comment:
np: AFAIK, it is not common to use underscores in service, paths or interfaces name. For instance, it is `org.freedesktop.NetworkManager` or `org.freedesktop.PolicyKit1`. I would suggest using `org.opensuse.TransactionalUpdate`.

Reply:
Thanks, I overlooked that, there are different conventions for service name, interface name, object paths,....